src/main/java/emissary/client/HTTPConnectionFactory.java

```diff
@@ -183,16 +195,50 @@ private static KeyStore buildStore(final String path, final char[] pazz, final S
         if ((path == null) || path.isEmpty()) {
             return null;
         }
+        Path filePath = Paths.get(path);
         final KeyStore keyStore = KeyStore.getInstance(type);
-        try (final InputStream is = Files.newInputStream(Paths.get(path))) {
-            keyStore.load(is, pazz);
+        if (isPemCert(filePath)) {
+            byte[] pemBytes = Files.readAllBytes(filePath);
+            byte[] derBytes = parseDerFromPemCert(pemBytes);
+            X509Certificate x509Certificate = generateCertFromDer(derBytes);
+            keyStore.load(null, pazz);
+            keyStore.setCertificateEntry("alias", x509Certificate);
+        } else {
+            try (final InputStream is = Files.newInputStream(filePath)) {
+                keyStore.load(is, pazz);
+            }
         }
         return keyStore;
     }
 
+    protected static boolean isPemCert(Path filePath) throws IOException {
+        byte[] beginBytes;
+        try (DataInputStream reader = new DataInputStream(Files.newInputStream(filePath))) {
+            int totalNumBytes = reader.available();
+            if (totalNumBytes <= BEGIN_CERT.length() + END_CERT.length()) {
+                return false;
+            }
+            beginBytes = new byte[BEGIN_CERT.length()];
+            reader.read(beginBytes);
```

Comment:
nit:
Not sure how much this matters in the real world so will defer to you more senior devs. Sonar complains about not checking the return value of the `read` call.
Would a `reader.readFully` make more sense here?
```suggestion
            reader.readFully(beginBytes);
```

Reply:
I think this makes sense, unlikely to matter in practice but simpler code.

Reply:
Updated